lib/arbre/rails/template_handler.rb

```diff
@@ -1,3 +1,19 @@
+ActionView::Base.class_eval do
+  def capture(*args)
+    value = nil
+    buffer = with_output_buffer { value = yield(*args) }
+
+    # Override to handle Arbre elements inside helper blocks.
+    # See https://github.com/rails/rails/issues/17661
+    # and https://github.com/rails/rails/pull/18024#commitcomment-8975180
+    value = value.to_s if value.is_a?(Arbre::Element)
```

Comment:
Copied ActionView::Helpers::CaptureHelper.capture and added this line to convert Arbre::Element#to_s so the output is not lost.